dts/bindings/mtd/mxicy,mx25r-spi.yaml

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Properties for features only available on the Macronix Ultra Low Power
+  SPI NOR flash devices (MX25R*) using the Zephyr spi-nor flash driver.
+
+compatible: "mxicy,mx25r-spi"
```

Comment:
I had it without `-spi` at first, but I think we'll need to add a `-qspi` version in the future to add this feature to the qspi driver. If we add the bus now we won't break others code when we have to change this later. So I figured I might as well add it now.

Reply:
But can't we have one binding that could be used with SPI and QSPI? It seems that it could include "jedec,spi-nor-common.yaml" instead of ""jedec,spi-nor.yaml".